MQTT/native cloud ingest (#4203)
influxdata/docs-v2: content/influxdb/v2.3/write-data/no-code/load-data.md

To ingest MQTT (Message Queuing Telemetry Transport) data into InfluxDB, do the following to set up a Cloud native subscription:

1. [Subscribe to an MQTT topic](#subscribe-to-an-mqtt-topic) in InfluxDB Cloud by configuring an MQTT broker, and specifying the topic(s) to subscribe to.
2. [Define parsing rules](#define-parsing-rules) for JSON or regex data (line protocol requires no configuration).

{{% note %}}
For troubleshooting help, see [Troubleshoot MQTT ingest errors](/influxdb/cloud/write-data/troubleshoot/#troubleshoot-mqtt-ingest-errors/)
{{% /note %}}

#### Subscribe to an MQTT topic

1. In the navigation menu on the left, click **Load Data** > **Cloud Native Subscriptions**.
    {{< nav-icon "data" >}}
2. Click **+ Create Subscription**.
3. On the **Setting Up - MQTT Subscriber** page, under **Connect to Broker**, enter the following:
   - Subscription Name
   - Description
   - Protocol
   - Hostname or IP address (hostname or URL of the MQTT broker)
   - Port (TCP/IP port number the MQTT broker uses)
   - Security details. Choose one of the following: 
      - **None**
      - **Basic** (username/password)
      - **Certificate**

> 4. Under **Subscribe to a topic**, in the **Topic** field, enter the MQTT topic name to subscribe to. Note, MQTT brokers typically support the wild cards `+` and  `#`.

Reviewer: ```suggestion
4. Under **Subscribe to a topic**, in the **Topic** field, enter the MQTT topic name to subscribe to. Note, MQTT brokers typically support wildcard subscriptions with the wildcard characters `+` and  `#`. 
```

   - To subscribe to all topics in a directory, use `+`. For example, if an `iotdevices` directory includes two directories called `asia` and `europe`, to subscribe to a `sensor` topic in either directory, use `iotdevices/+/sensors` to subscribe to `iotdevices/asia/sensors`, and `iotdevices/europe/sensors`.

>    - To subscribe to all topics in a directory, use `#`. For example, `iotdevices/#` subscribes to all topics in the  `iotdevices` directory.

Reviewer: ```suggestion
   - To subscribe to all topics in a directory, use `#`. For example, `iotdevices/#` subscribes to all topics in the  `iotdevices` directory. For more information about MQTT subscriptions and wildcards, see [the MQTT specification for Topic Names and Topic Filters](https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901241).
```

   {{% warn %}}
   **Important**: Configure the timestamp format that matches the format in your messages.
   {{% /warn %}}

  2. Under **Measurement**, enter the **JSON path** (start with `$.`) to assign the InfluxDB measurement key, for example, `$.device_id` or `$.device_information.device_id` for a nested measurement key.
  3. Select the **Data Type** for the measurement.
  4. Specify the JSON paths to tag and field names as needed, and then select the data type for the tag or field. At least one field is required. Note, JSON paths with arrays are supported, for example, `$.device_information.errors_encountered[0].error_number`.

{{% /tab-content %}}

<!-------------------------------- BEGIN String -------------------------------->
{{% tab-content %}}

Associate **String** key/value pairs with **InfluxDB elements** (measurements, timestamps, fields, or tags).

1. On the **Setting Up - MQTT Connector** page, under **Data Format**, do the following:

>   1. (Optional) In the **Regex pattern to find timestamp** field, enter the regex to find the timestamp in the MQTT message.  Otherwise, InfluxDB automatically assigns a timestamp when messages are ingested into InfluxDB.

Reviewer: ```suggestion
  1. (Optional) In the **Regex pattern to find timestamp** field, enter the regex (regular expression) to find the timestamp in the MQTT message.  Otherwise, InfluxDB automatically assigns a timestamp when messages are ingested into InfluxDB.
```

{{% /tab-content %}}

<!-------------------------------- BEGIN String -------------------------------->
{{% tab-content %}}

To associate **String** key/value pairs with **InfluxDB elements** (measurements, timestamps, fields, or tags), do the following:

- On the **Setting Up - MQTT Connector** page, under **Data Format**, do the following:

  1. (Optional) In the **Regex pattern to find timestamp** field, enter the regex to find the timestamp in the MQTT message.  Otherwise, InfluxDB automatically assigns a timestamp when messages are ingested into InfluxDB.
  
      {{% note %}}
      **Note**: Parsing rules only support finding one value at a time.
      {{% /note %}}
      
      For example, if the timestamp string is `time=1653998899010000000`, use a regex to find the string you're looking for and capture the timestamp:
   - `time=([\s\S]*?)\n` (captures value after `=` until the EOL (end of line) is reached)
   - `time=([\s\S]*?),` (captures value after `=` until comma is reached)

>    - time=([\s\S]*?),

Reviewer 2: kelly left off here
